src/ucp/wireup/wireup_cm.c

```diff
@@ -793,6 +821,8 @@ static void ucp_cm_disconnect_cb(uct_ep_h uct_cm_ep, void *arg)
     ucs_trace("ep %p flags 0x%x: remote disconnect callback invoked", ucp_ep,
               ucp_ep->flags);
 
+    UCP_CM_CB_CHECK_UCP_EP(ucp_ep, uct_cm_ep, return);
```

Comment:
why is it ok to check ep flags w/o async lock?

Reply:
added check that async lock is blocked when checking this flag